## Scaling and generating gcode from `Toadstool Logo Scaleable 2.svg`

This is what Inkscape shows:

![image.png](./7f83b16.png)

so that's 657.39001 mm wide which is 25.8815 inches, which is pretty close to 24 already.

In [3]:
originalWidth = 657.39011 / 25.4
originalWidth

25.88150039370079

In [5]:
24 / originalWidth * 657.39011

609.6

so say 610 for grins.

Question is now how to scale it?

Inkscape has a "Scale" option on Document Properties -> Page, so computing the scaling factor is as follows:

In [6]:
610 /657.39011

0.9279117387391178

So this scale, when converted to a percentage may be put into Inkscape's Transform -> Scale tab as shown below:

![thing](./054999d.png)

I save this file as `Toadstool Logo Scaled to 24" wide.svg`

Now I need to figure out how to generate g-code for it using gcodetools.

According to Extensions -> Gcodetools -> About, the source for gcodetools may be found at http://www.cnc-club.ru/gcodetools

The available menu items in the version I have (1.7) are shown below:

![gcode tools menu](./2767639.png)

I've no idea where to start, so I've picked "Path to Gcode" and the following values on each of its three tabs ("Path to Gcode", "Options", "Preferences").  The fourth tab "Help" has no values.

![gcodetools Path to Gcode -> Path to Gcode](./2bf08ef.png)

![gcodetools Path to Gcode -> Options](./b26dc3d.png)

![gcodetools Path to Gcode -> Preferences](./681bc32.png)

When I hit Apply I got:

![Inkscape Error](./Screenshot_20240428_205710.png)

I added a layer (Layer 1) with nothing in it and hit Apply again:

![Inkscape Error](./Screenshot_20240428_205934.png)

However a file did appear:

output_001.ngc




In [1]:
cat output_0001.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z5.000000
G00 X586.714272 Y21.868317

G01 Z-0.125000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-0.125000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-0.125000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-0.125000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-0.125000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-0.125000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-0.125000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-0.125000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-0.125000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-0.125000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-0.125000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-0.125000 I8.004507 J19.089994
G02 X556.956

This line didn't seem quite right:

```
G01 Z-0.125000 F100.0(Penetrate)
```

That's 0.125 mm?

We need more like ${1 \over 4}$" which is 6.35 mm.

Not sure how to make that happen.  Maybe try Options -> "Scale along Z axis:" assuming that it's derived from "Minimum arc radius"?  It could have been generated from a default tool too I suppose.

Generated another .ngc file by hitting "Apply".

In [3]:
cat output_0002.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z255.000000
G00 X586.714272 Y21.868317

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-6.375000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-6.375000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-6.375000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-6.375000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-6.375000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-6.375000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-6.375000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-6.375000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-6.375000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-6.375000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-6.375000 I8.004507 J19.089994
G02 X556.9

This seemed to do the trick as far as the depth is concerned:

```
G01 Z-6.375000 F100.0(Penetrate)
```

Not sure if the Genmitsu 4030 can cut 6.375mm depth at 400mm/min though for the whole path.

Also, I still don't know if the depth value was derived from the "Minimum arc radius" or not.

To test this, I'll set "Minimum arc radius" to be 1mm instead of 0.125mm.

In [4]:
cat output_0003.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path916)
(Change tool to Default tool)

G00 Z255.000000
G00 X586.714272 Y21.868317

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.652674 Y21.374962 Z-6.375000 I2.515280 J5.983719 F400.000000
G03 X592.272452 Y21.868317 Z-6.375000 I-0.720401 J11.027769
G02 X594.400686 Y22.442614 Z-6.375000 I14.010297 J-47.688896
G02 X595.566533 Y22.350830 Z-6.375000 I0.444421 J-1.805171
G02 X596.687193 Y21.083930 Z-6.375000 I-0.745068 J-1.788164
G02 X596.577955 Y19.362960 Z-6.375000 I-2.519815 J-0.704008
G02 X595.138321 Y17.361025 Z-6.375000 I-5.094816 J2.145186
G02 X592.402360 Y15.289435 Z-6.375000 I-15.272390 J17.327847
G02 X581.936385 Y10.920297 Z-6.375000 I-17.039964 J26.098246
G02 X568.054002 Y12.208774 Z-6.375000 I-4.895285 J22.687369
G02 X556.651510 Y24.096796 Z-6.375000 I8.004507 J19.089994
G02 X556.9

No change, so it must be choosing a default tool for me, or getting the 0.125 default depth from somewhere else.

And this leads me to realize that for a ${1 \over 8}" diameter tool, the "Minimum arc radius:" must be 1.5875mm, so I set that as the default instead of 0.125mm which was crazy anyway (it was sort of in inches not mm).

Now the tool I selected was actually "cylinder", and it has parameters as defined below:

![cylindrical cutter parameters](./Screenshot_20240428_214631.png)

So these values are clearly not all correct.  The diameter of tool I'm going to use is ${1 \over 8}$" which is 3.175mm.  I'm not sure of the \[cutting?\] feed rate of 400 either.  The penetration feed of 100 might be okay, but I'm really not sure.  These need to be set up correctly for the Genmitsu 4030 and the bit I'm going to be using.

The gcodetools Tools dialog suggests all I need to do is modify the text values in the text box in the .svg file, and that'll change the output.  I guess that's worth a test.  I'll change the diameter and the feed and see what happens.

![changed cylindrical cutter parameters](./Screenshot_20240428_215434.png)


In [5]:
cat output_0004.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path842)
(Change tool to Cylindrical cutter)

G00 Z255.000000
G00 X586.325014 Y-264.838936

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.263423 Y-265.332293 Z-6.375000 I2.515286 J5.983732 F200.000000
G03 X591.883207 Y-264.838936 Z-6.375000 I-0.720403 J11.027793
G01 X595.177295 Y-264.356422 Z-6.375000
G01 X596.188719 Y-267.344298 Z-6.375000
G02 X594.749082 Y-269.346238 Z-6.375000 I-5.094827 J2.145190
G02 X592.013115 Y-271.417832 Z-6.375000 I-15.272423 J17.327884
G02 X581.547117 Y-275.786979 Z-6.375000 I-17.040001 J26.098302
G02 X567.664704 Y-274.498500 Z-6.375000 I-4.895295 J22.687418
G02 X556.262187 Y-262.610452 Z-6.375000 I8.004524 J19.090035
G02 X556.566876 Y-247.923100 Z-6.375000 I18.749258 J6.957883
G02 X573.063021 Y-222.573661 Z-6.375000 I78.032730 J-32.737638
G03 X601.153048 Y-179.39

That did seem to change the feed rate, but I can't tell how it changed the tool diameter.  I guess I'll just have to assume it worked also.

```
G03 X589.263423 Y-265.332293 Z-6.375000 I2.515286 J5.983732 F200.000000
```

I notice the "depth step" tool parameter also.  I wonder what would happen if I changed that to say 4.  Would that perform 4 iterations with successively deeper depths until -6.375mm is reached?

Changed it and Apply'd it.

In [6]:
cat output_0005.ngc

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path842)
(Change tool to Cylindrical cutter)

G00 Z255.000000
G00 X586.325014 Y-264.838936

G01 Z-6.375000 F100.0(Penetrate)
G03 X589.263423 Y-265.332293 Z-6.375000 I2.515286 J5.983732 F200.000000
G03 X591.883207 Y-264.838936 Z-6.375000 I-0.720403 J11.027793
G01 X595.177295 Y-264.356422 Z-6.375000
G01 X596.188719 Y-267.344298 Z-6.375000
G02 X594.749082 Y-269.346238 Z-6.375000 I-5.094827 J2.145190
G02 X592.013115 Y-271.417832 Z-6.375000 I-15.272423 J17.327884
G02 X581.547117 Y-275.786979 Z-6.375000 I-17.040001 J26.098302
G02 X567.664704 Y-274.498500 Z-6.375000 I-4.895295 J22.687418
G02 X556.262187 Y-262.610452 Z-6.375000 I8.004524 J19.090035
G02 X556.566876 Y-247.923100 Z-6.375000 I18.749258 J6.957883
G02 X573.063021 Y-222.573661 Z-6.375000 I78.032730 J-32.737638
G03 X601.153048 Y-179.39

No change. In other words no text differences between output_0004.ngc and output0005.ngc


So I set it back to 1.

Incidentally, we're at revision 21ff0ba6eae4febcb5393ad4d27af4c0f8ead7de now.

Okay. So assuming this gcode is somewhat accurate I can see I have a problem.

The logo is too big to be done on the Genmitsu 4030.  The 4030 has an X range of 400mm and a Y range of 300mm.

Eyeballing the dimensions of the logo, it appears to need something like 600mm for X range and 260mm for Y range.

This suggests I'll need to break up the job.

Looking at output_0005.ngc, I see that it seems to work on paths within the SVG such as path842. It appears that there are several blocks for each path.  path842 has two blocks at the beginning of the file.  These cut out the "l" in toadstool.

I suppose I could extract these gcode chunks and reassemble them in several separate files along with coordinate changes to make them fit within the available space of the Genmitsu.

I could also cut and paste the path objects (or several of them at a time) and position them to cut out of a piece of material in new .svg files, from which I could generate gcode files for each.

The latter seems like it might be more practical given my severely limited knowledge of gcode (and svg).

It looks like the "to" goes together as a contiguous chunk, and that the last "o" and "l" are each a contiguous chunk.

The "adsto" appears to be the biggest contiguous chunk.

So I'll break it up into two .svgs.  "Toadstool Logo Scaleable 2 to-ol.svg" and "Toadstool Logo Scaleable 2 adsto.svg"

Complete as of 2a3cd910245d9a08cf67fa53a1df28f3efdf0dcc.

I notice while working on "Toadstool Logo Scaleable 2 to-ol.svg" that the default "orientation points" placed on the drawing are as shown below:

![orientation points](./Screenshot_20240429_155902.png)

If this is a correct assumption, perhaps I can change the z-depth directly here rather than by using the scaling thing.

To test this guess, I will generate gcode for this file.  Before I do that, I change the default output file name to something more associable with "Toadstool Logo Scaleable 2 to-ol.svg", namely "Toadstool Logo Scaleable 2 to-ol.ngc"

"Apply" generates the file "Toadstool Logo Scaleable 2 to-ol_0001.ngc"

In [7]:
cat 'Toadstool Logo Scaleable 2 to-ol_0001.ngc'

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path842)
(Change tool to Cylindrical cutter)

G00 Z255.000000
G00 X242.205389 Y11.506217

G01 Z-6.375000 F100.0(Penetrate)
G03 X245.143795 Y11.012857 Z-6.375000 I2.515290 J5.983714 F200.000000
G03 X247.763579 Y11.506217 Z-6.375000 I-0.720381 J11.027602
G01 X251.057669 Y11.988727 Z-6.375000
G01 X252.069089 Y9.000847 Z-6.375000
G02 X250.629454 Y6.998914 Z-6.375000 I-5.094866 J2.145230
G02 X247.893489 Y4.927317 Z-6.375000 I-15.272397 J17.327803
G02 X237.427481 Y0.558170 Z-6.375000 I-17.039995 J26.098303
G02 X223.545069 Y1.846647 Z-6.375000 I-4.895294 J22.687467
G02 X212.142559 Y13.734696 Z-6.375000 I8.004542 J19.090040
G02 X212.447249 Y28.422047 Z-6.375000 I18.749256 J6.957882
G02 X228.943393 Y53.771489 Z-6.375000 I78.032717 J-32.737627
G03 X257.033419 Y96.948347 Z-6.375000 I-104.736549 J9

The resulting depth is as expected:

```
G03 X245.143795 Y11.012857 Z-6.375000 I2.515290 J5.983714 F200.000000
```

But now I will change -0.125 to -6.35, thus:

![change to -6.35](./Screenshot_20240429_160935.png)

"Apply" and generate "Toadstool Logo Scaleable 2 to-ol_0002.ngc"

In [8]:
cat 'Toadstool Logo Scaleable 2 to-ol_0002.ngc'

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path842)
(Change tool to Cylindrical cutter)

G00 Z255.000000
G00 X242.205389 Y11.506217

G01 Z-51.000000 F100.0(Penetrate)
G03 X245.143795 Y11.012857 Z-51.000000 I2.515290 J5.983714 F200.000000
G03 X247.763579 Y11.506217 Z-51.000000 I-0.720381 J11.027602
G01 X251.057669 Y11.988727 Z-51.000000
G01 X252.069089 Y9.000847 Z-51.000000
G02 X250.629454 Y6.998914 Z-51.000000 I-5.094866 J2.145230
G02 X247.893489 Y4.927317 Z-51.000000 I-15.272397 J17.327803
G02 X237.427481 Y0.558170 Z-51.000000 I-17.039995 J26.098303
G02 X223.545069 Y1.846647 Z-51.000000 I-4.895294 J22.687467
G02 X212.142559 Y13.734696 Z-51.000000 I8.004542 J19.090040
G02 X212.447249 Y28.422047 Z-51.000000 I18.749256 J6.957882
G02 X228.943393 Y53.771489 Z-51.000000 I78.032717 J-32.737627
G03 X257.033419 Y96.948347 Z-51.000000 I-

Predictably, the new Z-depth is wildly high:

```
G03 X245.143795 Y11.012857 Z-51.000000 I2.515290 J5.983714 F200.000000
```

So I'd better adjust the scaling:

![Scale along Z axis: 1](./Screenshot_20240429_161307.png)

And again "Apply"

In [9]:
cat 'Toadstool Logo Scaleable 2 to-ol_0003.ngc'

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path842)
(Change tool to Cylindrical cutter)

G00 Z5.000000
G00 X242.205389 Y11.506217

G01 Z-1.000000 F100.0(Penetrate)
G03 X245.143795 Y11.012857 Z-1.000000 I2.515290 J5.983714 F200.000000
G03 X247.763579 Y11.506217 Z-1.000000 I-0.720381 J11.027602
G01 X251.057669 Y11.988727 Z-1.000000
G01 X252.069089 Y9.000847 Z-1.000000
G02 X250.629454 Y6.998914 Z-1.000000 I-5.094866 J2.145230
G02 X247.893489 Y4.927317 Z-1.000000 I-15.272397 J17.327803
G02 X237.427481 Y0.558170 Z-1.000000 I-17.039995 J26.098303
G02 X223.545069 Y1.846647 Z-1.000000 I-4.895294 J22.687467
G02 X212.142559 Y13.734696 Z-1.000000 I8.004542 J19.090040
G02 X212.447249 Y28.422047 Z-1.000000 I18.749256 J6.957882
G02 X228.943393 Y53.771489 Z-1.000000 I78.032717 J-32.737627
G03 X257.033419 Y96.948347 Z-1.000000 I-104.736549 J98.

Okay, that's super-interesting, because now it is taking 1mm depth passes until 6.35mm is reached.  That implies that "depth step" of "1" mean "1mm per step"

I'm going to repeat these steps for 

This time however, I explicitly added Orientation points thus:

![create 2-points mode orientation points](./Screenshot_20240429_162130.png)

This produced an unexpected result on the .svg though:

![result of creating orientation points](./Screenshot_20240429_162305.png)

I corrected this on the .svg by hand:

![manually corrected](./Screenshot_20240429_162534.png)

And here's the generated gcode:

In [11]:
cat ./'Toadstool Logo Scaleable 2 adsto_0001.ngc'

%
(Header)
(Generated by gcodetools from Inkscape.)
(Using default header. To add your own header create file "header" in the output dir.)
M3
(Header end.)
G21 (All units in mm)

(Start cutting path id: path838)
(Change tool to Cylindrical cutter)

G00 Z5.000000
G00 X399.239687 Y35.660241

G01 Z-1.000000 F100.0(Penetrate)
G02 X393.010112 Y18.892945 Z-1.000000 I-63.176967 J13.931331 F200.000000
G02 X383.670483 Y7.489992 Z-1.000000 I-34.024509 J18.341590
G02 X371.154668 Y1.214759 Z-1.000000 I-19.783051 J23.837984
G02 X353.493625 Y1.039844 Z-1.000000 I-9.209330 J38.160668
G02 X335.210943 Y13.172490 Z-1.000000 I6.561252 J29.728558
G02 X330.815456 Y34.890074 Z-1.000000 I22.345097 J15.826091
G02 X346.384805 Y59.210667 Z-1.000000 I40.850694 J-9.007625
G02 X380.412352 Y66.680339 Z-1.000000 I25.096223 J-33.084172
G02 X396.496138 Y54.654746 Z-1.000000 I-5.239552 J-23.776239
G02 X399.239687 Y35.660241 Z-1.000000 I-23.843236 J-13.139285
G01 X399.239687 Y35.660241 Z-1.000000
G00 Z5.000000

(End cut

I think it's time to try this out.

We're at 8178e3afad91c1cde23475b879e08ddb27363a1f.

Actually we're at 2f75f1bf803208eb90275d728356e97b56d8eb0b, and upon examining the dimensions of 'Toadstool Logo Scaleable 2 adsto.svg' I notice that it will just barely fit (400.03528):

![document resized to path](./Screenshot_20240502_161945.png)

